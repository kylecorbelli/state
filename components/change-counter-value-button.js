import React, { PropTypes } from 'react';
import {
  Text,
  TouchableOpacity,
  StyleSheet
} from 'react-native';

const ChangeCounterValueButton = ({ text, color, onPress }) => (
  <TouchableOpacity
    style={[{backgroundColor: color}, styles.button]}
    onPress={onPress}
  >
    <Text style={styles.buttonText}>{text}</Text>
  </TouchableOpacity>
);

ChangeCounterValueButton.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  button: {
    width: 130,
    height: 70,
    justifyContent: 'center',
    alignItems: 'center',
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5
  },
  buttonText: {
    color: '#fffddb',
    fontSize: 50
  }
});

export default ChangeCounterValueButton;
