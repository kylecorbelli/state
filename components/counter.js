import React, { Component, PropTypes } from 'react';
import {
  View,
  ScrollView,
  RefreshControl,
  Text,
  StyleSheet
} from 'react-native';
import { connect } from 'react-redux';
import {
  increment,
  decrement,
  syncCloudCount
} from '../actions';
import ChangeCounterValueButton from './change-counter-value-button';

class Counter extends Component {
  static propTypes = {
    counter: PropTypes.object.isRequired,
    increment: PropTypes.func.isRequired,
    decrement: PropTypes.func.isRequired,
    syncCloudCount: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.syncCloudCount();
  }

  render() {
    const { counter, increment, decrement, syncCloudCount } = this.props;
    return (
      <ScrollView
        contentContainerStyle={styles.screen}
        refreshControl={
          <RefreshControl
            refreshing={counter.isSyncing}
            onRefresh={syncCloudCount}
          />
        }
      >
        <Text style={styles.content}>{counter.currentValue}</Text>
        <View style={styles.buttons}>
          <ChangeCounterValueButton
            text='-'
            color='#baa077'
            onPress={decrement}
          />
          <ChangeCounterValueButton
            text='+'
            color='#77dd79'
            onPress={increment}
          />
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: '#44c0ff'
  },
  content: {
    fontSize: 200,
    color: '#fffddb',
    fontFamily: 'arial'
  },
  buttons: {
    flexDirection: 'row',
    width: '90%',
    justifyContent: 'space-around',
    alignItems: 'center'
  }
});

const mapStateToProps = (state) => ({
  counter: state.counter
});

export { Counter };

export default connect(
  mapStateToProps,
  { increment, decrement, syncCloudCount }
)(Counter);
