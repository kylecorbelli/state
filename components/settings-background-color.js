import React from 'react';
import {
  View,
  Text
} from 'react-native';

const BackgroundColorSettings = () => (
  <View>
    <Text>Background Color Settings</Text>
  </View>
);

export default BackgroundColorSettings;
