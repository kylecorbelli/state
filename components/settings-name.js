import React from 'react';
import {
  View,
  Text
} from 'react-native';

const NameSettings = () => (
  <View>
    <Text>Name Settings</Text>
  </View>
);

export default NameSettings;
