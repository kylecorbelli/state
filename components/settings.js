import React from 'react';
import {
  View,
  Text
} from 'react-native';
import { List, ListItem } from 'react-native-elements';

const Settings = ({ navigation }) => (
  <View>
    <List>
      <ListItem
        title='Name'
        onPress={() => navigation.navigate('NameSettings')}
      />
      <ListItem
        title='Background Color'
        onPress={() => navigation.navigate('BackgroundColorSettings')}
      />
    </List>
  </View>
);

export default Settings;
