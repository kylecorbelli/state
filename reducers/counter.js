import { combineReducers } from 'redux';
import isSyncing from './counter-is-syncing';
import currentValue from './counter-current-value';

const counter = combineReducers({
  isSyncing,
  currentValue
});

export default counter;
