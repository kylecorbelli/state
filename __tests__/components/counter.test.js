import 'react-native';
import React from 'react';
import { Counter } from '../../components/counter';
import renderer from 'react-test-renderer';

describe('<Counter />', () => {
  it('should match Snapshot', () => {
    const tree = renderer.create(
      <Counter
        counter={6}
        increment={() => {}}
        decrement={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
