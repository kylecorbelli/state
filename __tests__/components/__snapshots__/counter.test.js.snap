// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should match Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#44c0ff",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fffddb",
        "fontFamily": "arial",
        "fontSize": 200,
      }
    }
  >
    6
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "90%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#baa077",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "height": 70,
          "justifyContent": "center",
          "opacity": 1,
          "width": 130,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fffddb",
            "fontSize": 50,
          }
        }
      >
        -
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#77dd79",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "height": 70,
          "justifyContent": "center",
          "opacity": 1,
          "width": 130,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fffddb",
            "fontSize": 50,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
